mesh_map/src/mesh_map.cpp

```diff
@@ -524,49 +514,40 @@ void MeshMap::combineVertexCosts(const rclcpp::Time& map_stamp)
 {
   RCLCPP_INFO_STREAM(node->get_logger(), "Combining costs...");
 
+  // instead: just store last layer
   float combined_min = std::numeric_limits<float>::max();
-  float combined_max = std::numeric_limits<float>::min();
+  float combined_max = std::numeric_limits<float>::lowest();
 
   vertex_costs = lvr2::DenseVertexMap<float>(mesh_ptr->nextVertexIndex(), 0);
 
+  auto [last_layer_name, last_layer] = loaded_layers.back();
+
+  const float default_value = last_layer->defaultValue();
+  const auto& costs = last_layer->costs();
   bool hasNaN = false;
-  for (auto layer : loaded_layers)
+
+  const float factor = 1.0;
+
+  for (auto vH : mesh_ptr->vertices())
   {
-    const auto& costs = layer.second->costs();
-    float min, max;
-    mesh_map::getMinMax(costs, min, max);
-    const float norm = max - min;
-    const float factor = 1.0;
-    // TODO: carefully think about this
-    const float norm_factor = factor / norm;
-
-    if(norm <= 0.00001)
+    const float cost = costs.containsKey(vH) ? costs[vH] : default_value;
+    if (std::isnan(cost))
     {
-      RCLCPP_ERROR_STREAM(node->get_logger(), "Layer \"" << layer.first << "\": ERROR - range between max and min value has to be >0.");
+      hasNaN = true;
     }
-
-    RCLCPP_INFO_STREAM(node->get_logger(), "Layer \"" << layer.first << "\" max value: " << max << " min value: " << min << " norm: " << norm
-                               << " factor: " << factor << " norm factor: " << norm_factor);
-
-    const float default_value = layer.second->defaultValue();
-    hasNaN = false;
-    for (auto vH : mesh_ptr->vertices())
+    vertex_costs[vH] += factor * cost;
```

Comment:
```suggestion
    vertex_costs[vH] = factor * cost;
```
Since we only look at one layer now, would this be equivalent? Note that `vertex_costs` is a class member, but I think it gets reset to all-zeroes in line `521`:
`vertex_costs = lvr2::DenseVertexMap<float>(mesh_ptr->nextVertexIndex(), 0);`

This is just minor polish, but might make the intention of the new functionality clearer.

Reply:
I agree. However, with the next changes I make, I think this will be removed again.